### ISSUE TITLE
device.discoverServices() returns null

### ISSUE THREAD

**Issue by PhilColbert** [pauldemarco/flutter_blue#495]
Hi,

I have about 30 known phones working perfectly with flutter blue, however a random chinese phone returns null on device.discoverServices()

Any ideas ?

Thanks

**Comment by Fleximex** [comment]
Does the phone itself have location services turned on or not the right bluetooth/location permission in the app?

**Comment by PhilColbert** [comment]
Yes, it connects to the ble device no problems, just trying to discover services, it fails.

Thanks

**Comment by pauldemarco** [comment]
Hi @PhilColbert ,

Can you share the phone model that is experiencing this?
